driver/driver.go

```diff
@@ -1,18 +1,32 @@
 package driver
 
+// SystemInfo gives more insight into system details
+type SystemDetails struct {
+	IsWindows bool
+	IsLinux   bool
+	IsDarwin  bool
+	IsWeb     bool
+	Name      string
+	Extra     string
+}
+
 type fields struct {
```

Comment:
you think we can make this something else or give it a more concrete name? like `driverBase`